src/main/java/com/google/research/bleth/servlets/EnqueueExperimentServlet.java

```diff
@@ -0,0 +1,233 @@
+import com.google.research.bleth.simulator.MovementStrategyFactory;
+import com.google.research.bleth.simulator.Schema;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/**
+ * A servlet used for enqueuing multiple tasks targeted at endpoint '/new-experiment-simulation',
+ * in order to create and run a new experiment.
+ */
+@WebServlet("/enqueue-experiment")
+public class EnqueueExperimentServlet extends HttpServlet {
+    private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private static final Gson gson = new Gson();
+    private static final MovementStrategyFactory.Type defaultMovementStrategy = MovementStrategyFactory.Type.RANDOM;
+    private static final AwakenessStrategyFactory.Type defaultAwakenessStrategy = AwakenessStrategyFactory.Type.RANDOM;
+    private static final String PROJECT_ID = "bleth-2020";
+    private static final String LOCATION_ID = "europe-west1";
+    private static final String QUEUE_ID = "simulations-queue";
+    private static final String queueName = QueueName.of(PROJECT_ID, LOCATION_ID, QUEUE_ID).toString();
+    private static final Logger log = Logger.getLogger(EnqueueExperimentServlet.class.getName());
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        if (request.getServerName().equals("localhost") && request.getServerPort() == 8080) {
+            response.setContentType("text/plain;");
+            response.getWriter().println("Experiments are not yet supported on localhost.");
+            return;
+        }
+
+        log.info("Enqueue Experiment Servlet invoked.");
+
+        String experimentTitle = request.getParameter("experimentTitle");
+        int beaconsNum = Integer.parseInt(request.getParameter("beaconsNum"));
+        Set<List<PropertyWrapper>> configurations = createConfigurations(request);
+
+        Entity experiment = new Entity(Schema.Experiment.entityKind);
+        experiment.setProperty(Schema.Experiment.experimentTitle, experimentTitle);
+        Key experimentId = datastore.put(experiment);
+        log.info("A new experiment entity with id " + KeyFactory.keyToString(experimentId) +
+                "was created and written to db.");
+
+        int legalConfigurationsCount = 0;
+        for (List<PropertyWrapper> configuration : configurations) {
+            if (validateArguments(configuration)) {
```

Comment:
Consider checking this out before, i.e. filtering the Cartesian product so that the assistant function returns only valid configurations. If you do so you don't have to count the new tasks here.